documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,387 @@
+- A static Lerp function which takes 2 Quaternions and a Scalar matching the generic type which linearly interpolates between the 2 Quaternions with scalar used as the amount to lerp
+- A static SLerp function which takes 2 Quaternions and a Scalar matching the generic type which Spherical linearly interpolates between the 2 Quaternions with scalar used as the amount to lerp
+- A static Zero Quaternion Property
+- A static Identity Quaternion Property
+
+# Geometric Types
+
+The following Geometric Types are defined:
+- BoxF
+- BoxI
+- CircleF
+- CircleI
+- PlaneF
+- PlaneI
+- Ray2F
+- Ray2I
+- Ray3F
+- Ray3I
+- RectangleF
+- RectangleI
+- SphereF
+- SphereI
+
+Each type **must** include the following:
+- Intersect functions with both another instance of the type and a point
+- GetDistanceToNearest(Point,Edge,etc) functions **must** be available for a given point
+- For all but the rays and planes, GetInflated function that takes a point and returns the scaled object that is closest to the original and contains the given point
+- Include Scale and Translation transformation functions
+- For Box and Rectangle the following Vector properties **must** be defined
+  - Min
+  - Max
+  - Center
+  - Size
+- For Planes and Rays, Normalize functions
+- For Planes include the following static functions
+  - CreateFromVerticies
+  - CreateFromPointNormal
+  - Dot
+    - with a Vector4
+  - DotCoordinate and DotNormal
+    - with a Vector3
+  - Transform
+    - With a Matrix4x4 or Quaternion, if relevant
+
+# Meeting Notes 
+
+## 19/11/2023
+
+[Video](https://www.youtube.com/live/yXNDZDE3AHE?feature=shared&t=9444)
+
+- We agree with the addition to add scalar operations over vectors (i.e. Vector4<T> * T does X*T, Y*T, ...)
+- Add an analyser for encouraging the most correct and most efficient type instead of using sub-optimal types. 
+- Ensure we've documented that.
+- Vector * Matrix?
+    - assuming row major, not allowing matrix * vector (only vector * matrix)
+    - does it make sense to just do it the way game engines do it and the way 2.0 does it?
+    - possibly technically invalid mathematically, even if common programmatically.
+    - easy to hit corner case that not all users may understand.
+- Ensuring we have no upcasting and downcasting operators that could have unexpected behaviour.
+- System.Numerics is strictly right hand, which is the same as XNA and somewhat default for OpenGL users. This differs with DirectX that is typically left hand.
+    - Happy with having explicit LH and RH functions.
+- Missing matrix APIs that were added in .NET 8 to System.Numerics
+    - CreateLookTo (additional to CreateLookAt)
+    - CreateViewport
+    - CreateBillboardLH
+    - CreateConstrainedBillboardLH
+    - CreateBillboardRH
+    - CreateConstrainedBillboardRH
+- Investigate newer less-hated alternatives to quaternions for future (see Freya Holmer talks)
```

Comment:
@Perksey don't suppose you have a link? Quaternions are, of course, ontologically evil, so I would be interested to hear about viable alternatives. :eyes:

Reply:
I think it was @HurricanKai or @ThomasMi that raised this point (and alluded to things like rotors as well)